internal/ingress/controller/store/store.go

```diff
@@ -132,13 +129,13 @@ type Informer struct {
 
 // Lister contains object listers (stores).
 type Lister struct {
-	Ingress           IngressLister
```

Comment:
Why do we still need this?

Reply:
We cannot get rid of this because the informer still needs this type.

Reply:
Does this mean we will be storing extensions.Ingress objects in both here and `IngressWithAnnotation`? That sounds like unnecessary duplication.

I'm not too familiar with how k8s go client works, but it does not make sense to not be able to configure informer without storing the objects. We should not need IngressLister, since all the ingress objects are already going to be stored in `IngressWithAnnotation`.

Reply:
> Does this mean we will be storing extensions.Ingress objects in both here and IngressWithAnnotation? That sounds like unnecessary duplication.

I know but there is no way to have a custom type (IngressWithAnnotation) in the informer.

> I'm not too familiar with how k8s go client works, but it does not make sense to not be able to configure informer without storing the objects

You need to use a store to be able to handle deltas.